MetaMask/metamask-extension
# fix: Prevent phishing detection network request when off or not onboarded

## **Description**

When the "Use phishing detection" feature in settings was toggled off, it was still possible for a network request for phishing related data to be made. That is because `this.phishingController.maybeUpdateState();` can result in a network request, and that was not gated by `usePhishDetect`. The same was true if the user was not yet onboarded. This PR corrects that.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24887?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Build this branch
2. Install and onboard
3. go through onboarding and toggle off "Use Phishing Detection"
4. You should not see network requests (in the background console) to phishing-detection.metafi.codefi.network
5. Open any dapp, you still should not see requests to that api
6. Go to settings and Toggle on "Use Phishing Detection"
7. Reload the extension and open any dapp
8. You should see a network request to phishing-detection.metafi.codefi.network

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
